pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -556,6 +560,12 @@ public boolean commitSegmentFile(String tableName, String segmentLocation, Strin
     File tableDir = new File(baseDir, tableName);
     File fileToMoveTo = new File(tableDir, segmentName);
 
+    if (!isConnected() || !isLeader()) {
+      LOGGER.warn("Lost leadership while committing segment file {}, {} for table {}", segmentName, segmentLocation,
```

Comment:
Log the isleader and isConnected status. Can be helpful while debugging edge cases

Reply:
Will do